polkadot-parachains/integritee-runtime/src/lib.rs

```diff
@@ -509,15 +479,154 @@ impl cumulus_pallet_parachain_system::Config for Runtime {
+where
+	AbsoluteMultiLocation: Get<MultiLocation>,
+{
+	fn reserve(asset: &MultiAsset) -> Option<MultiLocation> {
+		RelativeReserveProvider::reserve(asset).map(|relative_reserve| {
+			if relative_reserve == AbsoluteMultiLocation::get() {
+				MultiLocation::here()
+			} else {
+				relative_reserve
+			}
+		})
+	}
+}
+
+// We only support TEER for now.
+#[derive(
+	Encode,
+	Decode,
+	Eq,
+	PartialEq,
+	Copy,
+	Clone,
+	RuntimeDebug,
+	PartialOrd,
+	Ord,
+	TypeInfo,
+	MaxEncodedLen,
+)]
+pub enum CurrencyId {
+	TEER,
+}
+
+/// Converts a CurrencyId into a Multilocation, used by xtoken for XCMP.
+pub struct CurrencyIdConvert;
+impl Convert<CurrencyId, Option<MultiLocation>> for CurrencyIdConvert {
+	fn convert(id: CurrencyId) -> Option<MultiLocation> {
+		match id {
+			CurrencyId::TEER => Some(MultiLocation::new(
+				1,
+				X2(Parachain(ParachainInfo::parachain_id().into()), GeneralKey("TEER".into())),
+			)),
+		}
+	}
+}
+
+/// Converts a Mulitloaction into a CurrencyId. Used by XCMP LocalAssetTransactor for asset filtering:
+/// we only accept Assets that are convertable to a "CurrencyId".
+impl Convert<MultiLocation, Option<CurrencyId>> for CurrencyIdConvert {
+	fn convert(location: MultiLocation) -> Option<CurrencyId> {
+		let teer_key: Vec<u8> = "TEER".into();
+		let self_para_id: u32 = ParachainInfo::parachain_id().into();
+
+		match location {
+			MultiLocation { parents, interior } if parents == 1 => match interior {
+				X2(Parachain(para_id), GeneralKey(key))
+					if key == teer_key && para_id == self_para_id =>
+					Some(CurrencyId::TEER),
+				_ => None,
+			},
+			MultiLocation { parents, interior } if parents == 0 => match interior {
+				X1(GeneralKey(key)) if key == teer_key => Some(CurrencyId::TEER),
+				_ => None,
+			},
+			_ => None,
+		}
+	}
+}
+
+/// Converts a MultiAsset into a CurrencyId, using the defined Mulitlocation.
+impl Convert<MultiAsset, Option<CurrencyId>> for CurrencyIdConvert {
+	fn convert(asset: MultiAsset) -> Option<CurrencyId> {
+		if let MultiAsset { id: Concrete(location), .. } = asset {
+			Self::convert(location)
+		} else {
+			None
+		}
+	}
+}
+
+parameter_types! {
+	// This is how we are going to detect whether the asset is a Reserve asset
+	pub SelfLocation: MultiLocation = MultiLocation::here();
+	// We need this to be able to catch when someone is trying to execute a non-
+	// cross-chain transfer in xtokens through the absolute path way
+	pub SelfLocationAbsolute: MultiLocation = MultiLocation {
+		parents:1,
+		interior: Junctions::X1(
+			Parachain(ParachainInfo::parachain_id().into())
+		)
+	};
+}
+
+pub struct AccountIdToMultiLocation;
+impl Convert<AccountId, MultiLocation> for AccountIdToMultiLocation {
+	fn convert(account: AccountId) -> MultiLocation {
+		X1(AccountId32 { network: NetworkId::Any, id: account.into() }).into()
+	}
+}
+
+parameter_types! {
+	pub const BaseXcmWeight: Weight = 100_000_000; // TODO: recheck this
```

Comment:
I'd just check against Statemine.

Reply:
It's the same base weight as Acala uses. And they have this comment as well.. so I didn't want to remove it just like that.